### PR TITLE
added food comparator and corresponding tests

### PR DESCRIPTION
**Pull request by leezahuor**
# Issue/Task:
Implement logic for sorting `Food` objects by comparing self.expiration_date and self.name. 

When we call the comparator methods we expect:
- `Food` objects' `self.expiration_date` will be compared to `other.expiration_date`.
- Objects with earliest `self.expiration_date` will return earliest.
- If objects have the same `self.expiration_date`, they will be sorted alphabetically. 

# Checklist
- [X] Fixed Issue/Accomplished Task
- [X] Wrote tests
- [X] All test cases passing

### CODE REVIEW

File: tests/Utilities/constants.py [hunk 1]
```diff
@@ -18,6 +18,9 @@
 food1 = Food("apple", fooddate1.isoformat(), 3)
 food2 = Food("pear", fooddate2.isoformat(), 3)
 food3 = Food("banana", fooddate3.isoformat(), 2)
+food4 = Food("soda", fooddate1.isoformat(), 3)
+food5 = Food("apple", fooddate1.isoformat(), 3)
+food6 = Food("pumpkin", fooddate1.isoformat(), 3)
```
Comment (amandate) [inline tests/Utilities/constants.py:23]: Why did you need to create `food6` instead of just using `food4`?

**Review by amandate — COMMENTED** [review]
Just one question, but otherwise looks good!

File: tests/Utilities/constants.py (hunk 1 above)
Comment (leezahuor) [reply tests/Utilities/constants.py]: You're right, I got rid of `food6` and used `food4` instead.

**Review by amandate — APPROVED** [review]
LGTM good job!
